### ISSUE TITLE
Blockage a une etape pendant le calcule CrossTimeAnnotation(l1)

### ISSUE THREAD

**Issue by El-Castor** [yizhuo-wang/CASi#6]
Hello, 

First of all, thank you for this tool!

I wanted to try it on my data but when I apply the CrossTimeAnnotation() function each time I try it I get stuck at this stage and I don't know if this is normal because I don't get any error messages or warnings. Here is my terminal: 

....

Epoch 85/100
44/44 [==============================] - 0s 5ms/step - loss: 0.1938 - accuracy: 0.9430 - val_loss: 1.7952 - val_accuracy: 0.7511 - lr: 2.5000e-06
Epoch 86/100
44/44 [==============================] - 0s 5ms/step - loss: 0.1965 - accuracy: 0.9437 - val_loss: 1.7955 - val_accuracy: 0.7511 - lr: 2.5000e-06
Epoch 87/100
44/44 [==============================] - 0s 5ms/step - loss: 0.1850 - accuracy: 0.9447 - val_loss: 1.7959 - val_accuracy: 0.7511 - lr: 2.5000e-06
1302/1302 [==============================] - 2s 1ms/step

Following, the R script : 

# Charger les bibliothèques nécessaires
library(Seurat)
library(CASi)
library(tensorflow)
library(reticulate)

# Charger l'environnement Conda CASi
use_condaenv("CASi_Py310", required = TRUE)
#py_config()

# Définir les chemins
pre_seurat_file <- "/home/cpichot/work/Single_Cell_RNAseq/Bd21-3_ZE_timaline_scaleBio/downstream_analysis/INTEGRATED_seurat_obj/PRE_seurat_integrated_obj.rds"
output_dir <- "/home/cpichot/work/Single_Cell_RNAseq/Bd21-3_ZE_timaline_scaleBio/downstream_analysis/multiDataset_integration/CASi_out"
output_file <- file.path(output_dir, "l1.rds")

# Charger l'objet Seurat
seurat_obj <- readRDS(pre_seurat_file)

# Vérifier les échantillons disponibles
samples <- unique(seurat_obj$orig.ident)
print(samples)

# Définir les noms des échantillons
t0_sample <- "Bd21-ZE-D0.ScaleRNA"  # Temps initial
t1_samples <- c("Bd21-ZE-D1.ScaleRNA", "Bd21-ZE-D2.ScaleRNA", "Bd21-ZE-D3.ScaleRNA", 
                "Bd21-ZE-D4.ScaleRNA", "Bd21-ZE-D6.ScaleRNA")  # Autres temps

# Sélectionner les cellules des groupes
t0_cells <- colnames(seurat_obj)[seurat_obj$orig.ident == t0_sample]
t1_cells <- colnames(seurat_obj)[seurat_obj$orig.ident %in% t1_samples]

# Extraire les matrices de comptage
t0_count <- as.matrix(GetAssayData(seurat_obj, assay = "SCT", slot = "counts")[, t0_cells])
t1_count <- as.matrix(GetAssayData(seurat_obj, assay = "SCT", slot = "counts")[, t1_cells])

# Vérifier les dimensions
print(dim(t0_count))
print(dim(t1_count))

# Extraire les identités des cellules pour t0 (optionnel)
t0_label <- Idents(seurat_obj)[t0_cells]

# Créer l'objet CASI
l1 <- CreateCASiList(t0_count = t0_count, t1_count = t1_count, t0_label = t0_label)

# Exécuter l'annotation croisée avec TensorFlow optimisé pour GPU
l1 <- CrossTimeAnnotation(l1)

# Détecter les cellules nouvelles
l1 <- DetectNovelCells(l1)

# Sauvegarder les résultats
dir.create(output_dir, recursive = TRUE, showWarnings = FALSE)
saveRDS(l1, file = output_file)

print("Analyse terminée et fichier sauvegardé.")

Would you please have any suggestions for completing the analysis? 

Thank you in advance, 

Bests, 

Clement
